### DISCUSSION TITLE
Welcome to esp32_fm_synth Discussions!

### DISCUSSION THREAD

**Discussion by marcel-licence** [marcel-licence/esp32_fm_synth#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by matt-mcclintock** [comment]
Hi! I am having issues with setting up the esp32 Audio kit to work as the fm synth. I saw you combined the sampler project with the fm synth and followed your video as a guide. The issue I am having is that I cannot seem to get the device to work (I’m trying to test it with a virtual midi keyboard in cakewalk to see if everything works), there is no sound or an indication that it’s working.

**Reply by marcel-licence** [reply]
Hi, you could try to activate the NoteOn call in the setup function. That might be helpful to ensure that the device is able to produce sound. In addition to that a look on the debug output might also help. In case you have another Audio Kit variant as expected.

**Reply by matt-mcclintock** [reply]
I am still having issues. I followed along with your guide on setting up the esp32 audio kit v2.2 as a sampler but instead tried flashing the fm synth code. I have made sure that the libraries are up to date and I followed everything but it seems the device is still not functioning. I believe my model is the one with the ac101 audio dac as it does not have the gold antenna on the esp32 module. Is there any way you could help.

**Reply by marcel-licence** [reply]
In that case it looks like the ES8388 is on your board. As a first step you can look the the I2C scanner and use:
```
#define I2C_SDA 18
#define I2C_SCL 23

```
If the codec has been found `#define ES8388_ENABLED` should be enabled

In some case `#define ES8388_PIN_SCLK 5` should be replaced by `#define ES8388_PIN_SCLK 27`

I think I will also update this project soon to include the different board configuration files like in other projects.

**Reply by matt-mcclintock** [reply]
Okay, I will give this a try today. Thank you for all your help and making all of this available.

**Reply by marcel-licence** [reply]
Hi, I've updated the code. Now you will find a lot of known configurations in board_audio_kit_es8388.h. You can choose differen I2C and also different I2S settings. It looks like there are a lot of different configurations in use.
